microscopium/data/default-settings.yaml

```diff
@@ -0,0 +1,57 @@
+global:
+  # where to put the output file
+  output-dir: ''
+# The first phase is a feature computation from images specified in a csv file.
+# Input csv files should have one row per input image and minimally three
+# columns:
+#   - an *index* column containing a unique identifier for each image.
+#   - a *url* column containing the path to each image (can be absolute or
+#     relative to the csv file path).
+#   - a *replicate* column containing an identifier for each replicate, meaning
+#     images having received identical treatment.
+#   - optionally, *group* columns containing arbitrary groupings of samples.
+#     These will be used to color points in the plot and produce an interactive
+#     legend, so there should not be too many categories in any group.
+feature-computation:
+  source-table:
+    # column type : column name in dataset csv
+    index: index
+    path: url
+    replicates: reps
+    groups:
+      - group
+  feature-function:
+    function:
+      microscopium.feature.default_feature_map
+    kwargs:
+      random_seed : 0
+embeddings:
+  # Features produced by feature computation can be reduced to 2D for display
+  # as a scatter-plot. This is done with the below classes, which must follow
+  # the scikit-learn unsupervised `fit`/`fit_transform` API. The 0th component
+  # will map to x, and the 1st to y, and both will be placed in the
+  # corresponding column in the output.
+  UMAP:
+    class: umap.UMAP
+    x: x.umap
+    y: y.umap
+  tSNE:
+    class: sklearn.manifold.TSNE
+    x: x.tsne
+    y: y.tsne
+  PCA:
+    class: sklearn.decomposition.PCA
+    x: x.pca
+    y: y.pca
+  default: tSNE
+color-columns:
```

Comment:
color-columns may not be an very intuitive name, maybe we can think of a better one or document it really clearly